### ISSUE TITLE
feat: separate value from text content when filtering

### ISSUE THREAD

**Issue by Kinbaum** [pacocoursey/cmdk#74]
I have a use case where I want my value to be different than the label presented to the end user. When typing in the search field today, the user would have to know the underlying value to be able to see something return.... typing what is visible text content wouldn't yield the expected filtered results.

My ask is that we change the behavior so that the search field only looks at the text content rendered (we have to pass this anyway for something to show up). The only way an average user would know what the value is is if they opened their dev tools, which is not expected.

**Comment by raunofreiberg** [comment]
You should be able to pass a `value` prop to `Command.Item` and use `children` to display the label.

**Comment by Kinbaum** [comment]
Yes I understand this, but as I’ve pointed out, doing so affects the filtering capability. It will only filter on the value, not the children’s text content

**Comment by pacocoursey** [comment]
I think this is a valid use case that is a little difficult in the current API.

@Kinbaum Can you share where you want to use this internal-only value? Inside of `onSelect`?

---

As a workaround, you could include both values in the `value` string:

```tsx
<Command.Item value={`${internal}:${searchLabel}`}>{searchLabel}</Command.Item>
```

**Comment by Kinbaum** [comment]
I think so. And I’ve been stringifying a JSON object as sort of a hack (still has issues - see below) 

A use case would be pushing the user to a new route on select of an item. 

But this also poses challenges when setting the current value as that cannot be derived from the cleaned URL if you go with the hacked approach.

**Comment by Kinbaum** [comment]
@pacocoursey Is there any update on this feature request

**Comment by akd-io** [comment]
I have unique IDs for each of the items in my list, but because filtering happens on values and not labels, I'm having to use the labels as values, which is not ideal.

I wonder why filtering was done on values, to begin with. Isn't it always preferred to filter on labels?

**Comment by akd-io** [comment]
I think I was very confused in my last comment on here. I really wanted `value` to be a unique identifier, but I don't know why.

My requirements have changed a bit since. I now need to be able to find items with additional hidden search terms. I was able to get things working perfectly by appending additional search terms to the `value` property and using a custom filter function.

**Comment by malay44** [comment]
One way could be to pass the label to the `Command.Item` `keyword` prop. This might help, but it has one drawback: the filter will also show the items with matching values.

**Comment by gentunian** [comment]
@akd-io I'm mostly on the same page with shadcn. I was expecting to set value like an object and the label to be the value to be used to search for, i.e.

```
...
<CommandItem value={someObjectData}>My descriptive text</CommandItem>
...
```

but looks like it's the other way around. Cmdk uses `value` (and `keywords`) to search and `value` can only be a string: https://github.com/pacocoursey/cmdk/blob/main/cmdk/src/index.tsx#L44

Kind of weird. Regardless the usage of shadcn, I don't see how you can set an object specifying the item you want to filter other than use the value to index some other object.
